frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-install.tsx

```diff
@@ -0,0 +1,304 @@
+      props: { className: tableColumnClasses[0] },
+    },
+    {
+      title: 'Role',
+      props: { className: tableColumnClasses[1] },
+    },
+    {
+      title: 'CPU',
+      props: { className: tableColumnClasses[2] },
+    },
+    {
+      title: 'Memory',
+      props: { className: tableColumnClasses[3] },
+    },
+    {
+      title: 'Capacity',
+      props: { className: tableColumnClasses[4] },
+    },
+    {
+      title: 'Devices',
+      props: { className: tableColumnClasses[5] },
+    },
+  ];
+};
+NodeTableHeader.displayName = 'NodeTableHeader';
+const configMapsForAllNodes = {};
+
+const getConfigMaps = () => {
+  k8sGet(ConfigMapModel, null, 'openshift-storage').then(configMaps => {
+    console.log(configMaps, 'configmaps')
+    configMaps.items.forEach(config => {
+      const nodeName = config.metadata.labels && config.metadata.labels['rook.io/node'];
+
+      if (typeof nodeName !== 'undefined') {
+        configMapsForAllNodes[nodeName] = (JSON.parse(config.data.devices)).length;
+      }
+
+    });
+
+    console.log(configMapsForAllNodes, 'maps');
+  });
+};
+
+//need to use this for selection table
+// const NodeTableRows = (componentProps) => {
+//   return _.map(componentProps.data, obj => {
+//     if (obj && obj.metadata) {
+//       const cells = NodeTableRow(obj);
+//       const uid = obj.metadata.uid;
+//       const selected = false;
+//       return {
+//         selected,
+//         cells,
+//         uid,
+//       };
+//     }
+//   });
+// };
+
+const NodeTableRow: React.FC<NodeTableRowProps> = ({ obj: node, index, key, style }) => {
+  const roles = getNodeRoles(node).sort();
+  const devicesCount = configMapsForAllNodes[node.metadata.name] || 0;
+
+  return (
+    <TableRow id={node.metadata.uid} index={index} trKey={key} style={style}>
+      <TableData className={tableColumnClasses[0]}>
+        <ResourceLink kind="Node" name={node.metadata.name} title={node.metadata.uid} />
+      </TableData>
+      <TableData className={tableColumnClasses[1]}>
+        {roles.length ? roles.join(', ') : '-'}
+      </TableData>
+      <TableData className={tableColumnClasses[2]}>
+        {node.status && node.status.capacity ? node.status.capacity.cpu : '-'}
+      </TableData>
+      <TableData className={tableColumnClasses[3]}>
+        {node.status && node.status.allocatable ? node.status.allocatable.memory : '-'}
+      </TableData>
+      <TableData className={tableColumnClasses[4]}>
+        {node.status && node.status.capacity ? node.status.capacity.memory : '-'}
+      </TableData>
+      <TableData className={tableColumnClasses[5]}>
+        {devicesCount} Selected
+      </TableData>
+    </TableRow>
+  );
+};
+
+NodeTableRow.displayName = 'NodeTableRow';
+type NodeTableRowProps = {
+  obj: K8sResourceKind;
+  index: number;
+  key?: string;
+  style: object;
+};
+
+export class CreateOCSServiceForm extends React.PureComponent<CreateOCSServiceFormProps> {
```

Comment:
Prefer functional components with hooks instead of class components in new code

Reply:
@gnehapk You can still manage state in your component without using the class syntax.

Take `public/components/utils/alerts.tsx` as an example, it's a functional component (`React.FC`) that handles its state via `React.useState` API.

In general, we should prefer multiple smaller functional components over fewer bigger components, favoring composition over extension.